proof/access-control/RISCV64/ArchFinalise_AC.thy

```diff
@@ -229,10 +227,11 @@ lemma arch_finalise_cap_respects[wp]:
              intro: pas_refined_Control_into_is_subject_asid)
   done
 
-crunch arch_post_cap_deletion
-  for pspace_aligned[Finalise_AC_assms, wp]: "\<lambda>s :: det_ext state. pspace_aligned s"
-  and valid_vspace_objs[Finalise_AC_assms, wp]: "\<lambda>s :: det_ext state. valid_vspace_objs s"
-  and valid_arch_state[Finalise_AC_assms, wp]: "\<lambda>s :: det_ext state. valid_arch_state s"
+declare prepare_thread_delete_st_tcb_at_halted[Finalise_AC_assms]
+declare finalise_cap_valid_list[Finalise_AC_assms]
+declare arch_finalise_cap_pas_refined[Finalise_AC_assms]
+declare prepare_thread_delete_pas_refined[Finalise_AC_assms]
+declare finalise_cap_replaceable[Finalise_AC_assms]
```

Comment:
This is a bit awkward (not that the previous crunch was great either). Do we have statements of these somewhere? Because if yes, the declaration should be there so that we know at the statement/proof that they are interface lemmas. If they come out of different crunches now, we should move the corresponding declarations to those crunches.

Reply:
There's already these sorts of declares in quite a few arch files. My guess is that it is generally from when one architecture proves the lemma earlier, before the relevant locale and named theorem exists, which means that the crunch in this file doesn't do anything.

Alternatively, it could be a sign that the interface is too strong and that some of the lemmas could be removed from the locale, although we'd have to be careful to check all of the architectures to try that. In this case I'm a bit curious about whether the lemmas about generic functions could be removed, although that might not be worth exploring in this PR.

Reply:
It's alright, let's leave it.